source/Serial/NetworkSerial.cpp

```diff
@@ -75,132 +75,148 @@ NetworkSerial::~NetworkSerial(
 }
 
 /// \ref https://social.msdn.microsoft.com/Forums/windowsapps/en-US/961c9d61-99ad-4a1b-82dc-22b6bd81aa2e/error-c2039-close-is-not-a-member-of-windowsstoragestreamsdatawriter?forum=winappswithnativecode
 void
 NetworkSerial::end(
-	void
-	)
+    void
+    )
 {
-	_connection_ready = false;
-	_current_load_operation = nullptr;
-	_current_store_operation = nullptr;
-
-	// Reset with respect to dependencies
-	delete( _rx );
-	_rx = nullptr;
-	delete( _tx );
-	_tx = nullptr;
-	delete( _stream_socket );
-	_stream_socket = nullptr;
+    _connection_ready = false;
+    _current_load_operation = nullptr;
+    _current_store_operation = nullptr;
+
+    // Reset with respect to dependencies
+    delete( _rx );
+    _rx = nullptr;
+    delete( _tx );
+    _tx = nullptr;
+    delete( _stream_socket );
+    _stream_socket = nullptr;
 }
 
 void
 NetworkSerial::flush(
-	void
-	)
+    void
+    )
 {
-	_current_store_operation = _tx->StoreAsync();
+    _current_store_operation = _tx->StoreAsync();
+    create_task( _current_store_operation )
+        .then( [ this ]( unsigned int value_ )
+    {
+        return _tx->FlushAsync();
+    } )
+        .then( [ this ]( task<bool> task_ )
+    {
+        try
+        {
+            task_.get();
+
+            //detect disconnection
+            if( _current_store_operation->Status == Windows::Foundation::AsyncStatus::Error )
+            {
+                _connection_ready = false;
+                ConnectionLost( L"A fatal error has occurred in NetworkSerial::flush() and your connection has been lost." );
+            }
+        }
+        catch( Platform::Exception ^e )
+        {
+            _connection_ready = false;
+            ConnectionLost( L"A fatal error has occurred in NetworkSerial::flush() and your connection has been lost. Error: " + e->Message );
+        }
+    } );
 }
 
 uint16_t
 NetworkSerial::read(
-	void
-	)
+    void
+    )
 {
-	uint16_t c = static_cast<uint16_t>( -1 );
-
-	if( !connectionReady() ) {
-		return c;
-	}
-
-	if( available() ) {
-		c = _rx->ReadByte();
-	}
-	else if( _current_load_operation->Status != Windows::Foundation::AsyncStatus::Started ) {
-		// Attempt to detect disconnection
-		if( _current_load_operation->Status == Windows::Foundation::AsyncStatus::Error )
-		{
-			_connection_ready = false;
-			ConnectionLost();
-			return -1;
-		}
-
-		_current_load_operation = _rx->LoadAsync( 100 );
-	}
-
-	return c;
+    uint16_t c = static_cast<uint16_t>( -1 );
+
+    if( !connectionReady() ) {
+        return c;
+    }
+
+    if( available() ) {
+        c = _rx->ReadByte();
+    }
+    else if( _current_load_operation->Status != Windows::Foundation::AsyncStatus::Started ) {
+        // Attempt to detect disconnection
+        if( _current_load_operation->Status == Windows::Foundation::AsyncStatus::Error )
+        {
+            _connection_ready = false;
+            ConnectionLost( L"A fatal error has occurred in NetworkSerial::read() and your connection has been lost." );
```

Comment:
This string is used lots through the code. Pull into a constant - better than copy paste reuse.